src/common/regions.ts

```diff
@@ -0,0 +1,146 @@
+import { chain, fromPairs } from 'lodash';
+import urlJoin from 'url-join';
+import US_STATE_DATASET from 'components/MapSelectors/datasets/us_states_dataset_01_02_2020.json';
+import countyAdjacencyMsa from './data/county_adjacency_msa.json';
+const { state_dataset, state_county_map_dataset } = US_STATE_DATASET;
+
+export type FipsCode = string;
+
+export enum RegionType {
+  COUNTY = 'County',
+  STATE = 'State',
+  MSA = 'MSA',
+}
+
+abstract class Region {
+  constructor(
+    public name: string,
+    public urlName: string,
```

Comment:
for some reason my first thought is that url name would be the full relative url - I'm not sure if this is something other people think also or if it's just me :)

Reply:
The URL segment is loaded from the JSON file, it's easier to set the segment and construct the URL than to update the JSON file

Reply:
oh right.  maybe `urlSegment` actually? but i'm probably overthinking it :)

Reply:
I'm going to make it private, `urlSegment` is a bit better I think